internal/adminapi/client.go

```diff
@@ -67,6 +67,20 @@ func (c *Client) BaseRootURL() string {
 	return c.adminAPIClient.BaseRootURL()
 }
 
+func (c *Client) NodeID(ctx context.Context) (string, error) {
+	data, err := c.adminAPIClient.Root(ctx)
+	if err != nil {
+		return "", fmt.Errorf("failed fetching Kong client root: %w", err)
+	}
+
+	nodeID, ok := data["node_id"].(string)
+	if !ok {
+		return "", errors.New("malformed Kong node ID found in Kong client root")
```

Comment:
It's a method of the Admin API client, so there's no type of node here. In case this fails, logging is handled one level above, in the `NodeAgent`: https://github.com/Kong/kubernetes-ingress-controller/blob/5ebdf4a9c922c7dd9aec96f3976de3d9b83d3cf2/internal/konnect/node_agent.go#L424

Reply:
🤔 but the level above logging doesn't get to log the `nodeID` right? We'd need to include it in here. My suggestion was that: since the type assertion for `string` fails then we might as well print the type.

But because the map lookup is clumped up with the type assertion it can be either of those 2 failing. Maybe in that case we split this into 2 statements?